FreeRTOS/FreeRTOS-Kernel
# Fix context switch when time slicing is off

Description
-----------
When time slicing is off, context switch should only happen when a task with priority higher than the currently executing one is unblocked. Earlier the code was invoking a context switch even when a task with priority equal to the currently executing task was unblocked. This commit fixes the code to only do a context switch when a higher priority task is unblocked.

Test Steps
-----------
Create 2 tasks with same priority:
1. Task T1 - Creates task T2 and then blocks itself by calling `vTaskDelay`.
2. Task T2 - Never blocks.

```
void TaskT1( void *pvParams )
{
    xTaskCreate( prvTaskT2, "t2", 512, NULL, 1, NULL );

    for( ;; )
    {
        vTaskDelay(pdMS_TO_TICKS(1000));
    }
}
/*-----------------------------------------------------------*/
void TaskT2( void *pvParams )
{
    volatile uint32_t ulCounter = 0;

    for( ;; )
    {
        ulCounter++;
    }
}
/*-----------------------------------------------------------*/
````
When time slicing is off, the task T1 must never be scheduled again after it blocks.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
